docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -0,0 +1,382 @@
+{% set short_lang = 'python' %}
+{% if language == 'JavaScript' %}
+    {% set short_lang = 'js' %}
+{% endif %}
+
+{% set lowercase_lang = 'python' %}
+{% if language == 'JavaScript' %}
+    {% set lowercase_lang = 'javascript' %}
+{% endif %}
+
+***********************************************
+Transaction Processor Tutorial  {{ language }}
+***********************************************
+
+Overview
+========
+
+This tutorial covers the creation of a new Sawtooth Lake transaction family in
+{{ language }}, based on the Sawtooth Lake SDK. We will construct a transaction
+handler which implements a distributed version of the multi-player game tic-
+tac-toe.
+
+.. note::
+
+    The SDK contains a fully-implemented version of tic-tac-toe. This tutorial
+    is meant to demonstrate the relevant concepts, rather than to create a
+    complete implementation. See the SDK_ for full implemenations in
+    multiple languages.
+
+.. _SDK: https://github.com/hyperledger/sawtooth-core/tree/master/sdk/examples
+
+A general description of tic-tac-toe, including the rules, can be found on
+Wikipedia at:
+
+    https://en.wikipedia.org/wiki/Tic-tac-toe
+
+A full implementation of the tic-tac-toe transaction family can be found in
+
+``/project/sawtooth-core/sdk/examples/xo_{{ lowercase_lang }}/``.
+
+Prerequisites
+=============
+
+This tutorial assumes that you have gone through :doc:`/app_developers_guide/getting_started` and are
+familiar with the concepts introduced there.
+
+Prior to going through this tutorial, you should have a working Vagrant
+environment running to which you can login.  Specific setup instructions are
+available in :doc:`/app_developers_guide/getting_started`.
+
+The Transaction Processor
+=========================
+
+There are two top-level components of a transaction processor: a processor
+class and a handler class. The SDK provides a general-purpose processor class.
+The handler class is application-dependent and contains the business logic for
+a particular family of transactions. Multiple handlers can be connected to an
+instance of the processor class.
+
+Handlers get called in two ways:
+
+# An ``apply`` method
```

Comment:
https://build.sawtooth.me/job/Sawtooth-Hyperledger/job/sawtooth-core/job/PR-422/6/artifact/docs/build/html/_autogen/sdk_TP_tutorial_python.html#the-transaction-processor